src/main/java/seedu/address/logic/commands/EditClientCommand.java

```diff
@@ -109,13 +108,18 @@ public CommandResult execute(Model model) throws CommandException {
         if (personToEdit.getCar() != null) {
             Car carToEdit = personToEdit.getCar();
             editedCar = createEditedCar(carToEdit, editCarDescriptor);
-            if (carToEdit.equals(editedCar) && model.hasCar(editedCar) && isCarEdited) {
+            if ((carToEdit.equals(editedCar) || model.hasCar(editedCar)) && isCarEdited) {
                 throw new CommandException(MESSAGE_DUPLICATE_CAR);
             }
         }
 
         Person editedPerson = createEditedPerson(personToEdit, editPersonDescriptor, editedCar);
-        if (personToEdit.equals(editedPerson) && model.hasPerson(editedPerson) && isPersonEdited) {
```

Comment:
Is it better to just change the && to || instead of introducing more LOC?

i.e. `((personToEdit.equals(editedPerson) || model.hasPerson(editedPerson)) && isPersonEdited)`

Reply:
No i think it is better to separate them out

Reply:
I think it is better to seperate it out so that we can throw a different exception